### PR TITLE
refactor: remove zlib-sync

### PR DESCRIPTION
**Pull request by didinele**
**Please describe the changes this PR makes and why it should be merged:**
draft because vlad wants us to keep zlib-sync & the new pure-native node:zlib impl together for a few dev releases

**Status and versioning classification:**

### CODE REVIEW

File: packages/ws/src/ws/WebSocketShard.ts [hunk 1]
```diff
@@ -450,28 +454,21 @@ export class WebSocketShard extends AsyncEventEmitter<WebSocketShardEventsMap> {
...
-
-			const zlib = (await getZlibSync())!;
-			this.inflate.push(Buffer.from(decompressable), flush ? zlib.Z_SYNC_FLUSH : zlib.Z_NO_FLUSH);
-
-			if (this.inflate.err) {
-				this.emit(WebSocketShardEvents.Error, {
-					error: new Error(`${this.inflate.err}${this.inflate.msg ? `: ${this.inflate.msg}` : ''}`),
-				});
-			}
+				decompressable.length &&
```
Comment (kyranet) [inline packages/ws/src/ws/WebSocketShard.ts:458]:
```suggestion
				decompressable.length > 4 &&
```

**Review by kyranet — CHANGES_REQUESTED** [review]
Just a tiny thing:

File: packages/ws/src/ws/WebSocketShard.ts [hunk 2]
```diff
@@ -482,7 +479,7 @@ export class WebSocketShard extends AsyncEventEmitter<WebSocketShardEventsMap> {
 		this.debug([
 			'Received a message we were unable to decompress',
 			`isBinary: ${isBinary.toString()}`,
-			`useIdentifyCompress: ${this.useIdentifyCompress.toString()}`,
+			`useIdentifyCompress: ${this.useIdentifyCompression.toString()}`,
```
Comment (SuperchupuDev) [inline packages/ws/src/ws/WebSocketShard.ts:482]:
```suggestion
			`useIdentifyCompression: ${this.useIdentifyCompression.toString()}`,
```

**Comment by JMTK** [comment]
Just a small followup to this issue:
I created a PR for `zlib-sync` that addresses it not compiling on NodeJS 21 https://github.com/abalabahaha/zlib-sync/pull/11

**Comment by didinele** [comment]
closing in favor of a new PR since this has the branch on my fork and it's just a whole mess to work with.

**Pull request closed without merging**